src/ai/defaultai.cc

```diff
@@ -3330,18 +3331,31 @@ void DefaultAI::check_flag_distances(const Time& gametime) {
 // Dealing with diplomacy actions
 void DefaultAI::diplomacy_actions(const Time& gametime) {
 
+	const Widelands::PlayerNumber mypn = player_number();
+	const Widelands::Player* me = game().get_player(mypn);
+
+	// if we are defeated or the last one in a team leave team
+	if (me->team_number() != 0 &&
+	    (player_statistics.members_in_team(me->team_number()) == 1 || me->is_defeated())) {
```

Comment:
I've just checked that `is_defeated()` is expensive, so all checks should use the cached value from`player_statistics`.

Reply:
done